src/osdc/ObjectCacher.h

```diff
@@ -287,9 +287,7 @@ class ObjectCacher {
     void set_object_locator(object_locator_t& l) { oloc = l; }
 
     bool can_close() const {
-      if (lru_is_expireable()) {
-	ceph_assert(data.empty());
-	ceph_assert(waitfor_commit.empty());
+      if (lru_is_expireable() && data.empty() && waitfor_commit.empty()) {
```

Comment:
this PR actually changes the behavior. @dragonylffly why do you think we should have this change?

Reply:
its caller, for example ObjectCacher::release, does not ensure data.empty(), in addition, from the perspective of the literal meaning of the function, it should just tell the object _can__ close or not, leave its caller for the further job

Reply:
i am not familiar with this area. but have you read 011d1e79ab0c690bf2cf8cdb3f87c903e0ef4b9d . your change partially reverts it.

Reply:
yes